src/geometry/GeometryUtils.h

```diff
@@ -3,7 +3,10 @@
 #include "linalg.h"
 #include <vector>
 
-typedef std::vector<Vector3d> Polygon;
+class Polygon : public std::vector<Vector3d>  {
```

Comment:
In general, it's a bad idea to subclass STL containers. While this particular use-case may be safe, it's going to be easy to overlook if we further extend this class. Composition is probably a better solution.